test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1

```diff
@@ -554,6 +554,79 @@ Describe "Additional tests for Import-Module with WinCompat" -Tag "Feature" {
         }
     }
 
+    Context "Tests around Windows PowerShell Compatibility NoClobber module list" {
+        BeforeAll {
+            $pwsh = "$PSHOME/pwsh"
+            Add-ModulePath $basePath
+            $ConfigPath = Join-Path $TestDrive 'powershell.config.json'
+        }
+
+        AfterAll {
+            Restore-ModulePath
+        }
+
+        It "NoClobber WinCompat import works for an engine module through command discovery" {
+
+            ConvertFrom-String -InputObject '1,2,3' -Delimiter ',' | Out-Null
+            $modules = Get-Module -Name Microsoft.PowerShell.Utility
+            $modules.Count | Should -Be 2
+            $proxyModule = $modules | Where-Object {$_.ModuleType -eq 'Script'}
+            $coreModule = $modules | Where-Object {$_.ModuleType -eq 'Manifest'}
+
+            $proxyModule.ExportedCommands.Keys | Should Contain "ConvertFrom-String"
+            $proxyModule.ExportedCommands.Keys | Should Not Contain "Get-Date"
+
+            $coreModule.ExportedCommands.Keys | Should Contain "Get-Date"
+            $coreModule.ExportedCommands.Keys | Should Not Contain "ConvertFrom-String"
+
+            $proxyModule | Remove-Module -Force
+        }
+
+        It "NoClobber WinCompat import works for an engine module through -UseWindowsPowerShell parameter" {
+
+            Import-Module Microsoft.PowerShell.Management -UseWindowsPowerShell
```

Comment:
Ideally we should also have a test using a module specification, given the code devoted to making that work

Reply:
Added test for ModuleSpecifications.